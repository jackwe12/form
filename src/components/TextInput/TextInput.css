/* .text-input input{
    background-image:url('../../image/exclamation-circle.png');
    background-position:180px;
    background-size:10px;
    background-repeat: no-repeat;

} */
