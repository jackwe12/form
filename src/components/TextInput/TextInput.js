import * as React from 'react';
import ValidationError from '../ValidationError/ValidationError';

function TextInput({ name, label, type, value, alert, setInputs, placeholder }) {
    return (
        <div>
            <label htmlFor={name} className="uk-form-label uk-text-large">{label}</label>
            <input id={name} placeholder={placeholder} name={name} type={type || ""} value={value} onChange={setInputs} className={"uk-input" + (alert ? ' uk-form-danger' : '')} />
            <ValidationError content={alert} />
        </div>
    )
}

export default TextInput;