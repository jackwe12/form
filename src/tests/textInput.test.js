import TextInput from '../components/TextInput/TextInput';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
configure({ adapter: new Adapter() });


const mockHandler = jest.fn();
const wrapper = shallow(
    <TextInput 
        setInputs={mockHandler}
        {...{
        name: "lastName",
        label: "Last Name",
        type: "Text",
        placeholder: "Smith",
        alert: "",
        }
    }

/>)
const wrapper2 = shallow(
    <TextInput 
        setInputs={mockHandler}
        {...{
        name: "lastName",
        label: "Last Name",
        placeholder: "Smith",
        alert: "Error",
        }
    }

/>)
// const findByTestAttr = (wrapper, val) =>{
//     return wrapper.find(`[data-test="${val}]`);
// }

describe('Testing <TextInput/>', ()=>{
    test('label tag renders correct text',()=>{
        // expect(wrapper).find('label').hasClass('lastName').toBe(true);
        expect(wrapper.find('label').text()).toEqual('Last Name');
    });
    test('input tag render correctly prop',()=>{
        expect(wrapper.find('input').prop('name')).toEqual('lastName');
        expect(wrapper.find('input').prop('id')).toEqual('lastName');
        expect(wrapper.find('input').prop('placeholder')).toEqual('Smith');

    });
    test('should call mock changeHandler when input changes',()=>{
        const input = wrapper.find('input');
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
    test('when alert is empty, no uk-form-danger class name',()=>{
        expect(wrapper.find('input').hasClass("uk-form-danger")).toBe(false);       
    });
    test('when alert exists, uk-form-danger class name exists',()=>{
        expect(wrapper2.find('input').hasClass("uk-form-danger")).toBe(true);       

    });
});
