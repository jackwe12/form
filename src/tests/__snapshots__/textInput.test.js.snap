// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <TextInput/> snapshot testing 1`] = `
<div
  className="input-component text-input"
>
  <label
    className="form-label"
  >
    Last Name
  </label>
  <input
    className="uk-input uk-form-danger"
    id="lastName"
    name="lastName"
    onChange={[MockFunction]}
    placeholder="Smith"
    type=""
  />
  <div
    className="uk-text-danger"
    style={
      Object {
        "height": "21px",
      }
    }
  >
    <img
      alt="exclamation-circle.png"
      className="exclamation-circle"
      onClick={[Function]}
      src="exclamation-circle.png"
    />
    
  </div>
</div>
`;
